### DISCUSSION TITLE
`ensUniversalResolver` for L2s

### DISCUSSION THREAD

**Discussion by sergical** [wevm/viem#2211] [Question] [answered]
Is that something that will be available for chains like `OP Mainnet` and `Base`?

**Comment by wottpal** [comment, marked as answer]
ENS itself hasn't deployed resolvers on those chains yet as the preferred way of resolving is doing it via L1. Still, some domains' state (resolution records or subnames) has already moved to L2 via OffchainResolving & EVM Gateway (e.g. check out optinames.xyz).

**Reply by sergical** [reply]
Thank you for the context!
